volumes/ecocounter/tables/discontinuities.sql

```diff
@@ -0,0 +1,21 @@
+CREATE TABLE ecocounter.discontinuities (
+    uid serial PRIMARY KEY,
+    -- a references to EITHER a site_id or flow_id
+    site_id numeric REFERENCES ecocounter.sites (site_id),
+    flow_id numeric REFERENCES ecocounter.flows (flow_id),
+    -- moment the change takes place
+    break timestamp NOT NULL,
+    -- approximate bounds if the precise time is not known
+    give_or_take interval,
+    -- required description of what changed - be verbose!
+    notes text NOT NULL
+    CHECK (
+        -- only one or the other specified but not both
+        (site_id IS NOT NULL OR flow_id IS NOT NULL)
```

Comment:
The first clause is good enough for the checks, e.g.
```
-- does not return row `uid = 3`
SELECT *
FROM (VALUES 
	  (1, null, 1), 
	  (2, 2, null), 
	  (3, null, null)) as dates(uid, col_a, col_b)
where col_a IS NOT NULL OR col_b IS NOT NULL
```

Reply:
we could add a constraint for the check as well 
```
alter table ecounter.discontinuities
  add constraint discontinuities_site_id_flow_id
    check (site_id is not null or flow_id is not null);		
    ```

Reply:
> we could add a constraint for the check as well

As far as I know, that's just a more verbose way of doing the same thing. A `CHECK` in the initial table definition just _is_ a constraint.

Reply:
> The first clause is good enough for the checks

I wanted to prevent both `site_id` and `flow_id` from being defined at the same time though. One and only one of those should be defined, or else I'm not sure what it would mean.

Reply:
yea I think the only difference is we can name the constraint ourselves and since it will show up as a constraint in the table after creation, might as well include it when creating the table.

Reply:
Maybe we should just make both not null then ? I think was thinking if the discontinuities was discovered at one site_id, it will just cascade to all relevant flow_id within the site. But maybe we should just be very explicit in which flow_ids should be included in an array. This could help if the query only uses flow_id.

Reply:
Hmmm. I'll give this a little more thought. I can see now how having both options in this table is going to end up causing some confusion. Maybe we just define either the site or the flow level and also have a view that translates to the other level based on that.